src/TouchCarousel.js

```diff
@@ -198,6 +236,7 @@ class TouchCarousel extends React.PureComponent {
       modIndex += this.props.cardCount
     }
     this.props.onRest(index, modIndex, cursor, this.state)
+    this.modCursor()
```

Comment:
> being able to drag outside the carousel exposed a bug where the state was empty until the next start event...

I don't really get this. Can you explain more?
The moding was always done in touchStart event.

Reply:
With the mouse (and mouseMove and mouseUp events hooked to `document`) you can start dragging from the carousel and drag very far to the left or right, further than what padCards are filled. This means that the user sees empty state until the next time that `modCursor` is called. This doesn't happen with touch, because you can only drag to the end of the TouchCarousel, so the padCards are always enough.

Without this line, the user would be stuck in the empty state because `modCursor` was only called the next time they start drag. With this line, when the user stops dragging and the spring is at rest, we always call `modCursor` and the user sees the cards.

It would be event better, if the user never saw the empty state, but calling `modCursor` in the middle of user drag caused problems (drag got stuck), so this was a good enough solution.

Reply:
You can test this for yourself at http://localhost:5000/boring.html?loop&autoplay, drag with the mouse very fast and far.

Reply:
The `padCount` are designed for this. PadCards should always fill the carousel area no matter how far the user drags. If you see empty state, you need a bigger `padCount`.

Also, I think `modCursor()` in the spring rest event doesn't fully resolve your problem: the user has already seen the empty state before releasing the mouse.

Reply:
I increased the padCount in my for this reason, but still doing the `modCursor` here is good as defensive programming: if the user can get to the empty state, at least they don't get stuck.

Reply:
Defensive programming is for things that you cannot foresee, like drunk users' input and network errors. But in our case, the component's state is under fully control. We design and know exactly how this component behaves.

`modCursor()` in spring rest event may also cause other dilemmas. Say which cursor should we pass to the `props.onRest()`, the cursor before moding or after moding?